test/e2e/gatekeeper_controller_test.go

```diff
@@ -290,48 +291,49 @@ var _ = Describe("Gatekeeper", func() {
 
 			By("Getting gatekeeper")
 			Eventually(func() error {
-				return K8sClient.Get(ctx, types.NamespacedName{Name: "gatekeeper"},
-					gatekeeper)
-			}, timeout, pollInterval).ShouldNot(HaveOccurred())
+				return K8sClient.Get(ctx, types.NamespacedName{Name: "gatekeeper"}, gatekeeper)
+			}, timeout, pollInterval).Should(Succeed())
 
 			By("Getting Config")
 			config := &gkv1alpha1.Config{}
-			Eventually(func(g Gomega) error {
-				return K8sClient.Get(ctx, types.NamespacedName{Namespace: gatekeeperNamespace, Name: "config"},
-					config)
-			}, 120, 5).ShouldNot(HaveOccurred())
+			Eventually(func(g Gomega) []gkv1alpha1.MatchEntry {
+				g.Expect(
+					K8sClient.Get(ctx, types.NamespacedName{Namespace: gatekeeperNamespace, Name: "config"}, config),
+				).Should(Succeed())
 
-			By("Apply Config has 'shoudnotexist' ns")
-			Eventually(func() error {
-				config.Spec.Match = []gkv1alpha1.MatchEntry{
-					{
-						ExcludedNamespaces: []wildcard.Wildcard{
-							"shoudnotexist",
-						},
-						Processes: []string{
-							"webhook", "sync",
-						},
-					},
-				}
+				return config.Spec.Match
+			}, 150, 5).Should(HaveLen(2))
 
-				return K8sClient.Update(ctx, config)
-			}, timeout, pollInterval).ShouldNot(HaveOccurred())
+			By("Apply Config with 'shouldnotexist' namespace")
+			config.Spec.Match = []gkv1alpha1.MatchEntry{
+				{
+					ExcludedNamespaces: []wildcard.Wildcard{
+						"shouldnotexist",
+					},
+					Processes: []string{
+						"webhook", "sync",
+					},
+				},
+			}
+			// There is a gap after creation where an update doesn't cause a reconcile in the config_controller.
+			time.Sleep(30 * time.Second)
```

Comment:
Does removing the `CreateFunc` predicate on the config controller solve this? I'm wondering if there's a race condition where the config controller hasn't started before the update so it misses the update and doesn't do the initial reconcile on startup because of the predicate.

Reply:
Based on what we discussed, I've used the DynamicClient to instantiate the Config and then let the Config controller do the update to set the exempt namespaces. I ran the tests a couple times and it seems to behave as we'd expect, without the extra cache messages. I could move the Config `setExemptNamespaces()` over to be a method of the ConfigReconciler, but then I'd have to refactor the unit tests, which I didn't really want to get into.